kvirt/k3s/__init__.py

```diff
@@ -105,25 +112,39 @@ def create(config, plandir, cluster, overrides):
             del data['name']
         config.plan(plan, inputfile='%s/masters.yml' % plandir, overrides=data)
     firstmasterip, firstmastervmport = _ssh_credentials(k, firstmaster)[1:]
-    with open("%s/join.sh" % clusterdir, 'w') as f:
-        if api_ip is None:
-            api_ip = k.info(firstmaster)['ip']
-        joincmd = "curl -sfL https://get.k3s.io | K3S_URL=https://%s:6443 K3S_TOKEN=%s" % (api_ip, token)
-        if data['sdn'] != "flannel":
-            joincmd += " INSTALL_K3S_EXEC='--no-flannel'"
-        f.write("%s sh -\n" % joincmd)
-    source, destination = "/root/kubeconfig", "%s/auth/kubeconfig" % clusterdir
-    scpcmd = scp(firstmaster, ip=firstmasterip, user='root', source=source, destination=destination,
-                 tunnel=config.tunnel, tunnelhost=config.tunnelhost, tunnelport=config.tunnelport,
-                 tunneluser=config.tunneluser, download=True, insecure=True, vmport=firstmastervmport)
-    os.system(scpcmd)
+    ################
+    # WORKER SETUP #
+    ################
     workers = data.get('workers', 0)
     if workers > 0:
```

Comment:
the join.sh script needs to be generated even when no workers are specified initially, to allow to scale after the fact